src/plugins/GUITestBase/src/GTUtilsSequenceView.cpp

```diff
@@ -421,71 +421,62 @@ QString GTUtilsSequenceView::getSeqName(HI::GUITestOpStatus &os, ADVSingleSequen
             }
         }
     }
-    GT_CHECK(anns.size() != 0, QString("Annotation with name %1 and startPos %2").arg(name).arg(startpos));
-    GT_CHECK(anns.size() == 1, QString("Several annotation with name %1 and startPos %2. Number is: %3").arg(name).arg(startpos).arg(anns.size()));
+    GT_CHECK(selectedAnnotationList.size() != 0, QString("Annotation with annotationName %1 and startPos %2").arg(annotationName).arg(annotationRegionStartPos));
+    GT_CHECK(selectedAnnotationList.size() == 1, QString("Several annotation with annotationName %1 and startPos %2. Number is: %3").arg(annotationName).arg(annotationRegionStartPos).arg(selectedAnnotationList.size()));
 
-    Annotation *a = anns.first();
+    Annotation *annotation = selectedAnnotationList.first();
 
-    const SharedAnnotationData &aData = a->getData();
+    const SharedAnnotationData &aData = annotation->getData();
     AnnotationSettingsRegistry *asr = AppContext::getAnnotationsSettingsRegistry();
     AnnotationSettings *as = asr->getAnnotationSettings(aData);
 
-    const U2Region &visibleRange = seq->getDetView()->getVisibleRange();
-    QVector<U2Region> regions = a->getLocation().data()->regions;
     U2Region annotationRegion;
-    int regionId = 0;
-    foreach (const U2Region &reg, regions) {
-        if (reg.startPos == startpos - 1) {
+    int annotationRegionIndex = 0;
+    for (const U2Region &reg : annotation->getRegions()) {
+        if (reg.startPos == annotationRegionStartPos - 1) {
             annotationRegion = reg;
             break;
         }
-        regionId++;
+        annotationRegionIndex++;
     }
     GT_CHECK(!annotationRegion.isEmpty(), "Region not found");
 
-    if (!annotationRegion.intersects(visibleRange)) {
+    if (!annotationRegion.intersects(detView->getVisibleRange())) {
         int center = annotationRegion.center();
         goToPosition(os, center);
-        GTGlobals::sleep();
     }
 
-    const U2Region visibleRegionPart = annotationRegion.intersect(visibleRange);
-
-    U2Region y;
-    y = det->getAnnotationYRange(a, regionId, as);
-
-    float visibleRegionPartStart = visibleRegionPart.startPos;
-    float visibleRegionPartEnd = visibleRegionPart.endPos();
-    if (seq->getDetView()->isWrapMode() && annotationRegion.endPos() > visibleRange.endPos()) {
-        visibleRegionPartEnd = visibleRegionPart.startPos + seq->getDetView()->getSymbolsPerLine();
+    U2Region yRegion = renderArea->getAnnotationYRange(annotation, annotationRegionIndex, as);
+    U2Region visibleRegion = detView->getVisibleRange();
+    U2Region annotationVisibleRegion = annotationRegion.intersect(visibleRegion);
+    int x1 = renderArea->posToCoord(annotationVisibleRegion.startPos, true);
+    int x2 = renderArea->posToCoord(annotationVisibleRegion.endPos() - 1, true) + renderArea->getCharWidth();
+    if (x2 <= x1) {    // In the wrap mode x2 may be on a different line. In this case use [x1...line-end] as the click region.
+        x2 = detView->width();
```

Comment:
DetView is wider than DetViewRenderArea because of left vertical toolbar. There should be ```renderArea->width()```.

Reply:
Fixed.